File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/Pack.java
package com.shuneault.netrunnerdeckbuilder.game;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

/**
 * Created by sebast on 02/07/16.
 */

public class Pack {


    public void setName(String name) {
        this.name = name;
    }

    public void setCycleCode(String code) {
        this.cycle_code = code;
    }

    public static class SetCode {
        public static final String CORE_SET = "core";
        public static final String REVISED_CORE_SET = "core2";
        public static final String SYSTEM_CORE_2019 = "sc19";
    }

    public static final String KEY_CODE = "code";
    public static final String KEY_CYCLE_CODE = "cycle_code";
    public static final String KEY_DATE_RELEASE = "date_release";
    public static final String KEY_NAME = "name";
    public static final String KEY_POSITION = "position";
    public static final String KEY_SIZE = "size";

    private String code;
    private String cycle_code;
    private String date_release;
    private String name;
    private int position;
    private int size;
    private ArrayList<CardLink> cardLinks = new ArrayList<>();

    public Pack() {
    }

    public Pack(JSONObject json) {
        this.code = json.optString(KEY_CODE, "");
        this.cycle_code = json.optString(KEY_CYCLE_CODE, "");
        this.date_release = json.optString(KEY_DATE_RELEASE, "");
        this.name = json.optString(KEY_NAME, "");
        this.position = json.optInt(KEY_POSITION, 0);
        this.size = json.optInt(KEY_SIZE, 0);
        JSONArray cards = json.optJSONArray("cards");
        if (cards != null) {
            for (int i = 0; i < cards.length(); i++) {
                try {
                    JSONObject card;
                    card = cards.getJSONObject(i);
                    String cardCode = card.getString("code");
                    int quantity = card.getInt("quantity");
                    CardLink cl = new CardLink(cardCode, quantity);
                    this.cardLinks.add(cl);
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    public String getCode() {
        return code;
    }


    public void setCode(String code) {
        this.code = code;
    }

    public String getCycleCode() {
        return cycle_code;
    }

    public String getDateRelease() {
        return date_release;
    }

    public String getName() {
        return name;
    }

    public int getPosition() {
        return position;
    }

    public int getSize() {
        return size;
    }

    public ArrayList<CardLink> getCardLinks() {return cardLinks; }

    boolean isCoreSet() {
        return code.equals(SetCode.CORE_SET)
                || code.equals(SetCode.REVISED_CORE_SET)
                || code.equals(SetCode.SYSTEM_CORE_2019);
    }
}

File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/Pack.kt
package com.shuneault.netrunnerdeckbuilder.game

import org.json.JSONException
import org.json.JSONObject
import java.io.Serializable

/**
 * Created by sebast on 02/07/16.
 */
class Pack: Serializable {
    object SetCode {
        const val CORE_SET = "core"
        const val REVISED_CORE_SET = "core2"
        const val SYSTEM_CORE_2019 = "sc19"
    }

    var code: String? = null
    var cycleCode: String? = null
    var dateRelease: String? = null
        private set
    var name: String? = null
    var position = 0
        private set
    var size = 0
        private set
    val cardLinks = ArrayList<CardLink>()

    constructor() {}
    constructor(json: JSONObject) {
        this.code = json.optString(KEY_CODE, "")
        cycleCode = json.optString(KEY_CYCLE_CODE, "")
        dateRelease = json.optString(KEY_DATE_RELEASE, "")
        name = json.optString(KEY_NAME, "")
        position = json.optInt(KEY_POSITION, 0)
        size = json.optInt(KEY_SIZE, 0)
        val cards = json.optJSONArray("cards")
        if (cards != null) {
            for (i in 0 until cards.length()) {
                try {
                    var card: JSONObject
                    card = cards.getJSONObject(i)
                    val cardCode = card.getString("code")
                    val quantity = card.getInt("quantity")
                    val cl = CardLink(cardCode, quantity)
                    cardLinks.add(cl)
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
        }
    }

    val isCoreSet: Boolean
        get() = code == SetCode.CORE_SET || code == SetCode.REVISED_CORE_SET || code == SetCode.SYSTEM_CORE_2019

    companion object {
        const val KEY_CODE = "code"
        const val KEY_CYCLE_CODE = "cycle_code"
        const val KEY_DATE_RELEASE = "date_release"
        const val KEY_NAME = "name"
        const val KEY_POSITION = "position"
        const val KEY_SIZE = "size"
    }
}